src/biz/workspace/publish_dup.rs

```
@@ -740,52 +744,82 @@ impl PublishCollabDuplicator {
-          //                     // we need to replace them with respective duplicated row ids
-          //                 ],
-          //                 "field_type": Number(10),
-          //                 // use this condition to filter out relation cells
-          //             },
-          //         },
+          // "cells": Object {
+          //     "MBaTsr": Object {
+          //         "data": Array [
+          //             String("eefb5700-8cf7-411e-9596-f60b9a51916e"),
+          //             String("23d5e054-42c8-4754-ad69-527e4ffc1e46"),
+          //             // above are published row ids of related database
+          //             // we need to replace them with respective duplicated row ids
+          //         ],
+          //         "field_type": Number(10),
+          //         // use this condition to filter out relation cells
           //     },
-          // }
-          let cells: MapRef = db_row_collab
-            .data
-            .get_with_path(&txn, ["data", "cells"])
+          // },
+          let cells: MapRef = db_row_body
+            .get_data()
+            .get(&txn, ROW_CELLS)
             .ok_or_else(|| {
               AppError::RecordNotFound("no cells found in database row collab".to_string())
-            })?;
+            })?
+            .cast()
+            .map_err(|e| AppError::Unhandled(format!("not a map: {:?}", e)))?;
 
+          // collect all cell with field type as relation
           let mut rel_row_idss = vec![];
           for (_, out) in cells.iter(&txn) {
             if let Ok(m) = out.cast::<MapRef>() {
-              if let Some(Out::Any(Any::BigInt(n))) = m.get(&txn, "field_type") {
+              if let Some(Out::Any(Any::BigInt(n))) = m.get(&txn, CELL_FIELD_TYPE) {
                 if n == FieldType::Relation as i64 {
-                  let relation_data = m.get(&txn, "data").ok_or_else(|| {
+                  let relation_data = m.get(&txn, CELL_DATA).ok_or_else(|| {
```

deep copy database will be aborted when there is error. Are these errors can be ignored?

---

this shouldn't happen as this is a fundamental assumption of the data in the doc. nonetheless, I'll ignore it and log the error